File: app/src/main/java/com/esoxjem/movieguide/details/DetailsModule.java
package com.esoxjem.movieguide.details;

import com.esoxjem.movieguide.favorites.FavoritesInteractor;
import com.esoxjem.movieguide.network.RequestHandler;

import dagger.Module;
import dagger.Provides;

/**
 * @author pulkitkumar
 * @author arunsasidharan
 */
@Module
public class DetailsModule
{
    @Provides
    @DetailsScope
    MovieDetailsInteractor provideInteractor(RequestHandler requestHandler)
    {
        return new MovieDetailsInteractorImpl(requestHandler);
    }

    @Provides
    @DetailsScope
    MovieDetailsPresenter providePresenter(MovieDetailsInteractor detailsInteractor,
                                           FavoritesInteractor favoritesInteractor)
    {
        return new MovieDetailsPresenterImpl(detailsInteractor, favoritesInteractor);
    }
}

File: app/src/main/java/com/esoxjem/movieguide/movies/details/DetailsModule.kt
package com.esoxjem.movieguide.movies.details

import com.esoxjem.movieguide.movies.favorites.FavoritesContract
import com.esoxjem.movieguide.movies.favorites.FavoritesInteractor
import com.esoxjem.movieguide.network.TmdbWebService

import dagger.Module
import dagger.Provides

/**
 * @author pulkitkumar
 * @author arunsasidharan
 */
@Module
class DetailsModule {
    @Provides
    @DetailsScope
    fun provideInteractor(tmdbWebService: TmdbWebService): MovieDetailsContract.Interactor {
        return MovieDetailsInteractor(tmdbWebService)
    }

    @Provides
    @DetailsScope
    fun providePresenter(detailsInteractor: MovieDetailsContract.Interactor,
                         favoritesInteractor: FavoritesContract.Interactor): MovieDetailsContract.Presenter {
        return MovieDetailsPresenter(detailsInteractor, favoritesInteractor)
    }
}
